tools/Makefile

```
@@ -1,14 +1,29 @@
-all:	directory-cache
+.PHONY: check-password
 
-authority_certificate:
-	@echo "Please copy authority_* files to this directory"
-	exit 1
+AUTH_DIR=authority-private
+DIR_CACHE=directory-cache
 
-directory-cache: authority_certificate pip_install
-	mkdir -p directory-cache
-	./gen_fresh_dirinfo.py generate-dirinfo --consensus directory-cache/consensus.txt --microdescriptors directory-cache/microdescriptors.txt
+all: directory-cache
+
+directory-cache: pip_install check-password
+	mkdir -p $(DIR_CACHE)
+	mkdir -p $(AUTH_DIR)
+	./gen_fresh_dirinfo.py generate-certificate \
+	--authority-identity-key $(AUTH_DIR)/authority_identity_key \
+	--authority-signing-key $(AUTH_DIR)/authority_signing_key \
+	--authority-certificate $(DIR_CACHE)/certificate.txt \
+	--authority-v3ident $(DIR_CACHE)/authority.txt
+	./gen_fresh_dirinfo.py generate-dirinfo \
+	--authority-signing-key $(AUTH_DIR)/authority_signing_key \
+	--authority-certificate $(DIR_CACHE)/certificate.txt \
+	--consensus $(DIR_CACHE)/consensus.txt \
+	--microdescriptors $(DIR_CACHE)/microdescriptors.txt
```

indentation really helps with readability
```suggestion
	./gen_fresh_dirinfo.py generate-certificate \
		--authority-identity-key $(AUTH_DIR)/authority_identity_key \
		--authority-signing-key $(AUTH_DIR)/authority_signing_key \
		--authority-certificate $(DIR_CACHE)/certificate.txt \
		--authority-v3ident $(DIR_CACHE)/authority.txt
	./gen_fresh_dirinfo.py generate-dirinfo \
		--authority-signing-key $(AUTH_DIR)/authority_signing_key \
		--authority-certificate $(DIR_CACHE)/certificate.txt \
		--consensus $(DIR_CACHE)/consensus.txt \
		--microdescriptors $(DIR_CACHE)/microdescriptors.txt
```

also, as the arguments are always the same, you can merge both lines as to avoid duplication

---

```
for cmd in generate-certificate generate-dirinfo; do
  ./gen_fresh_dirinfo.py $cmd \
    ...
done
```

---

This actually drop errors, add a `set -e` before the loop to take care of that